docs/parameters.md

```diff
@@ -0,0 +1,324 @@
+      # passed as a JSON object to the broker
+      type: json 
+      value: >
+          {
+            "attributes": [
+                {
+                  "type": "a",
+                  "A": "1",
+                },
+                {
+                  "type": "b",
+                  "B": "2",
+                },
+            ],
+            "labels": ["foo", "bar"]
+          }
+```
+
+If the JSON object is stored in a `Secret` or a `ConfigMap` key, the combination of 
+`valueFrom` and `type: json` fields allows to declare this as well:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is a valid JSON which should be 
+      # passed as a JSON object to the broker
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: credentials
+```
+
+#### Other primitive types (int, bool, float)
+
+```yaml
+  ...
+  parameters:
+    - name: text
+      value: 1 # string by default, will be sent as "1"
+    - name: number
+      type: int
+      value: 1 # will be sent without quotes
+    - name: flag
+      type: bool
+      value: true
+    - name: fl
+      type: float
+      value: 123.5
+```
+
+#### Arrays
+A separate field `array: true` can be used to declare a value as an array. The
+element type is defined by the `type` value as described above.
+
+```yaml
+  ...
+  parameters:
+    - name: tags
+      array: true
+      value:
+      - foo
+      - bar
+    - name: jsonArray
+      type: json
+      array: true
+      value: [{"a": 1, "b": 2}]
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from 
+a particular source, Service Catalog also supports passing multiple parameters at 
+once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
```

Comment:
@duglin I agree. I have several options to discuss at the next design call:
- make a separate section apart from `parameters` and `parametersFrom`, aka `parametersBlob` with the options of: raw inline json, secretKeyRef, and configMapKeyRef with blob. Instead of having lists, I would provide a single field there, no merging strategies etc - just raw data, ready to send to the broker. I like that it explicitly states that this option doesn't do any magic at all, and the user must prepare the entire payload by himself, and it's not mixed with the other flexible options.
- add raw inline JSON, `secretKeyBlobRef` and `configMapBlobRef` to `parametersFrom`. I don't like that this could make it confusing and too complicated, but technically it is going to work.
- add `inline: true` attribute to the `parameters` section as an alternative to the named parameter. looks a bit hacky, but doesn't require adding more union types.